ext/net/ops.rs

```diff
@@ -484,9 +487,23 @@ where
 
   let resolver = AsyncResolver::tokio(config, opts)?;
 
-  resolver
-    .lookup(query, record_type)
-    .await
+  let lookup_fut = resolver.lookup(query, record_type);
+
+  let lookup = if let Some(cancel_rid) = cancel_rid {
+    let cancel_handle = state
+      .borrow_mut()
+      .resource_table
+      .get::<CancelHandle>(cancel_rid);
+    if let Ok(cancel_handle) = cancel_handle {
+      lookup_fut.or_cancel(cancel_handle).await?
```

Comment:
https://github.com/denoland/deno/pull/17736 showed a way to handle the cancel resource deletion entirely on the Rust side without needing to call another `tryClose` op. Preferably align with that.